### PR TITLE
Feature/proguard

### PR DESCRIPTION
**Pull request by ganfra**
This PR activates proguard in release to shrink and optimize code.
It doesn't activate obfuscation as it makes it hard then to read crash reports. 
Please try with care.

### CODE REVIEW

File: matrix-sdk-android/proguard-rules.pro [hunk 1]
```diff
@@ -19,3 +19,45 @@
 # If you keep the line number information, uncomment this to
 # hide the original source file name.
 #-renamesourcefileattribute SourceFile
+
+
+### EVENT BUS ###
```
Comment (bmarty) [inline matrix-sdk-android/proguard-rules.pro:24]: I like when config is split into several files

**Comment by bmarty** [comment]
Please add a line in CHANGE, enabling proguard can lead to hard-to-find bug (because not reproduced on debug mode), it will help to get some clue about it :)

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/Membership.kt [hunk 2]
```diff
@@ -17,10 +17,12 @@
 package im.vector.matrix.android.api.session.room.model
 
 import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
 
 /**
  * Represents the membership of a user on a room
  */
+@JsonClass(generateAdapter = false)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/Membership.kt:25]: I assume it's necessary for PG, but why `false`? This is the first time we use `false` :)

File: vector/build.gradle [hunk 3]
```diff
@@ -192,8 +192,14 @@ android {
             resValue "bool", "debug_mode", "false"
             buildConfigField "boolean", "LOW_PRIVACY_LOG_ENABLE", "false"
 
-            minifyEnabled false
-            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+
+            postprocessing {
+                removeUnusedCode true
+                removeUnusedResources true
+                obfuscate false
```
Comment (bmarty) [inline vector/build.gradle:199]: Maybe comment here why we do not obfuscate

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/Membership.kt (hunk 2 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/model/Membership.kt:25]: It's the rule for proguard recommanded by Moshi devs. It's to make it clear it shouldn't be removed but we don't need adapter as it use the EnumJsonAdapter from Moshi.

File: vector/build.gradle (hunk 3 above)
Comment (bmarty) [reply vector/build.gradle:199]: I do it

**Pull request merged**